modules/random/include/hephaestus/random/random_object_creator.h

```diff
@@ -153,12 +153,19 @@ namespace internal {
 /// Generate a random string of characters, including special case characters and numbers.
 template <StringType T>
 [[nodiscard]] auto random(std::mt19937_64& mt, std::optional<size_t> fixed_size = std::nullopt,
-                          bool allow_empty = true) -> T {
+                          bool allow_empty = true, bool lower_characters_only = false) -> T {
```

Comment:
Options for string creation start to sum up. What about adding a `RandomStringConfig` to bundle them?

Reply:
in this way it is possible to have default values

Reply:
```
struct RandomStringConfig {
  std::optional<size_t> fixed_size = std::nullopt,
  bool allow_empty = true;
  bool lower_characters_only = false;
} RANDOM_STRING_CONFIG_DEFAULT; 

[[nodiscard]] auto random(std::mt19937_64& mt, RandomStringConfig config = RANDOM_STRING_CONFIG_DEFAULT) -> T {
```
The issue with default values is that you can only omit arguments in order, so you can e.g. not say `random(mt, use_only_lower_case_characters)`, so default values are limited. A config does not make things worse.

Reply:
By the way, this should be `lower_case_characters_only` or `force_lower_case_characters`.

Reply:
I would resolve this, and address this in a follow up when we may want to add more options